src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -313,9 +352,27 @@ private float addRow(
           RowBufferStats.getCombinedStats(statsMap.get(columnName), forkedColStats.getValue()));
     }
 
-    // Increment null count for column missing in the input map
+    // Increment null count for column and its sub-columns missing in the input map
     for (String columnName : Sets.difference(this.fieldIndex.keySet(), inputColumnNames)) {
-      statsMap.get(columnName).incCurrentNullCount();
+      if (clientBufferParameters.getIsIcebergMode()) {
+        if (subColumnFinder == null) {
+          throw new SFException(ErrorCode.INTERNAL_ERROR, "SubColumnFinder is not initialized.");
+        }
+        subColumnFinder
+            .getSubColumns(columnName)
+            .forEach(
+                subColumn -> {
+                  RowBufferStats stats = statsMap.get(subColumn);
+                  if (stats == null) {
+                    throw new SFException(
+                        ErrorCode.INTERNAL_ERROR,
+                        String.format("Column %s not found in stats map.", subColumn));
+                  }
+                  statsMap.get(subColumn).incCurrentNullCount();
```

Comment:
repeated access of concurrent dictionaries for the same key is unnecessary. You can directly do `stats.incCurrentNullCount()` here

Reply:
fixed.